Trackers/iou_tracker/iou.py

```diff
@@ -0,0 +1,210 @@
+
+        """
+        tracks = []
+        detect = []
+        if len(self.tracks)==0:
+            tracks=np.zeros((1,4))
+        else:
+            for trk in self.tracks:
+                tracks.append(list(self.convert2relative(trk.bbox)))
+            tracks=np.array(tracks,dtype=np.float32)
+        for det in detections:
+            detect.append(list(self.convert2relative(det)))
+        detect=np.array(detect,dtype=np.float32)
+        IOU_mat = self.get_iou_matrix(tracks, detect)
+        if(len(self.tracks)*len(detections) == 1 or len(self.tracks)*len(detections) == 0):
+            IOU_mat = np.reshape(IOU_mat,(1,1))
+        else: 
+            IOU_mat = np.reshape(IOU_mat,(len(self.tracks),len(detections)))
+
+        # Solve the maximizing the sum of IOU assignment problem using the
+        # Hungarian algorithm (also known as Munkres algorithm)
+        row , col = linear_assignment(-IOU_mat)  
+        print("row",row,"col",col)  
+        unmatched_trackers, unmatched_detections = [], []
+        for t, track in enumerate(self.tracks):
+            if(t not in row):
+                unmatched_trackers.append(t)
+                print("t")
+
+        for d, det in enumerate(detections):
+            if(d not in col):
+                unmatched_detections.append(d)
+                print("d")
+
+        matches = []
+    
+        # For creating trackers we consider any detection with an 
+        # overlap less than iou_thrd to signifiy the existence of 
+        # an untracked object
+        t=row
+        d=col
+        for i in range(len(row)):
+            if(IOU_mat[t[i],d[i]]<iou_thrd):
+                unmatched_trackers.append(t[i])
+                unmatched_detections.append(d[i])
+            else:
+                matches.append((t[i],d[i]))
+        
+        if(len(matches)==0):
+            matches = np.empty((0,2),dtype=int)
+        print("matches",matches)
+        return matches, unmatched_detections, unmatched_trackers       
+        
+    def convert2relative(self,bbox):
+        """
+        Helper function to calculate left,top,right,bottom co-ordinates.
+
+        Args
+        bbox : bounding box containing tuple of co-ordinates.
+
+        Returns
+        left,top,right,bottom co-ordinates.
+        """
+        x, y, w, h  = bbox
+        _height = 416
+        _width = 416
+        x, y, w, h = x/_width,y/_height ,w/_width,h/_height
+        image_h, image_w = 640,480
+        orig_left    = int((x - w / 2.) * image_w)
+        orig_right   = int((x + w / 2.) * image_w)
+        orig_top     = int((y - h / 2.) * image_h)
+        orig_bottom  = int((y + h / 2.) * image_h)
+
+        if (orig_left < 0): orig_left = 0
+        if (orig_right > image_w - 1): orig_right = image_w - 1
+        if (orig_top < 0): orig_top = 0
+        if (orig_bottom > image_h - 1): orig_bottom = image_h - 1
+
+        return tuple((orig_left, orig_top, orig_right, orig_bottom))
```

Comment:
Try to put the utility functions below the methods which are important to logic of the code ie put this function after `get_iou_matrix`.